### ISSUE TITLE
Chart prevents synchronisation with ArgoCD when using custom sync label

### ISSUE THREAD

**Issue by aaaaahaaaaa** [hashicorp/vault-helm#1013]
**Describe the bug**
When using a custom sync label for ArgoCD (e.g `argocd.argoproj.io/instance` as per the official recommendation [here](https://argo-cd.readthedocs.io/en/stable/faq/#why-is-my-app-out-of-sync-even-after-syncing)), then the app stays `out-of-sync` in ArgoCD, without any actual diff being reported.
Looking then at the manifests being provisioned, the `argocd.argoproj.io/instance` label is missing entirely. So, AFAICT, there seems to be a very weird behaviour with that chart in particular that prevent ArgoCD from injecting the label.

For the record, we have over 30 apps running on our cluster in a very consistent way with the same overall config, and `vault` is the only one showing the weird behaviour.

**To Reproduce**
1. Configure ArgoCD with: `application.instanceLabelKey: argocd.argoproj.io/instance`
2. Install chart
3. App stays `Out-of-sync` in ArgoCD

`vault-0` manifest (`argocd.argoproj.io/instance` is missing):
```yaml
apiVersion: v1                                                                                                                                                                                                      
kind: Pod                                                                                                                                                                                                           
metadata:                                                                                                                                                                                                           
  creationTimestamp: "2024-04-10T12:25:48Z"                                                                                                                                                                         
  generateName: vault-                                                                                                                                                                                              
  labels:                                                                                                                                                                                                           
    app.kubernetes.io/instance: vault                                                                                                                                                                               
    app.kubernetes.io/name: vault                                                                                                                                                                                   
    component: server                                                                                                                                                                                               
    controller-revision-hash: vault-6b6bb87557                                                                                                                                                                      
    helm.sh/chart: vault-0.28.0                                                                                                                                                                                     
    statefulset.kubernetes.io/pod-name: vault-0                                                                                                                                                                     
  name: vault-0                                                                                                                                                                                                     
  namespace: vault 
```

![image](https://github.com/hashicorp/vault-helm/assets/2808155/82ac1dab-1fa0-42a8-8742-e95ac65b5254)

Other useful info to include: vault pod logs, `kubectl describe statefulset vault` and `kubectl get statefulset vault -o yaml` output

**Expected behavior**
* App is fully in sync in ArgoCD

**Environment**
* Kubernetes version: GKE
* vault-helm version: `0.28.0`

Chart values:

```yaml
apiVersion: argoproj.io/v1alpha1
kind: Application
metadata:
  name: vault-chart
  namespace: argocd
  finalizers:
    - resources-finalizer.argocd.argoproj.io
spec:
  project: default
  source:
    chart: vault
    repoURL: https://helm.releases.hashicorp.com
    targetRevision: 0.28.0
    helm:
      releaseName: vault
      valuesObject:
        server:
          ha:
            enabled: true
        ui:
          enabled: true
          serviceNodePort: null
          externalPort: 8200
  destination:
    server: "https://kubernetes.default.svc"
    namespace: vault
  syncPolicy:
    automated:
      prune: true
    syncOptions:
      - CreateNamespace=true
  ignoreDifferences:
    - group: admissionregistration.k8s.io
      kind: MutatingWebhookConfiguration
      jqPathExpressions:
        - .webhooks[]?.clientConfig.caBundle

```

**Additional context**

**Comment by aaaaahaaaaa** [comment]
Hmm, `statefulset` does have the `argocd.argoproj.io/instance: vault-chart` annotation. Now I'm really confused.

**Comment by aaaaahaaaaa** [comment]
It's actually the ArgoCD `application` manifest that seems to be missing the injected annotation. I'm really confused why.

![image](https://github.com/hashicorp/vault-helm/assets/2808155/58ed452b-76f7-4898-a584-c0d88e688f91)

**Comment by aaaaahaaaaa** [comment]
The problem came from ArgoCD ([issue](https://github.com/argoproj/argo-cd/issues/15126)). And was resolved with `2.11`.
